R/checks.R

```diff
@@ -1033,4 +1033,109 @@ check_invalid_form2.1 <- function(form){
     ) %>%
     mutate(comment="Invalid or missing entries")
   return(problems)
+}
+
+#' Check that all entries in form3.1 are filled with valid values
+#' 
+#' @param form dataframe containing form3.1
+#' @return a dataframe containing problematic entries
+#' 
+#' @import tidyverse
+#' @export
+check_invalid_form3.1 <- function(form) {
+  problems <- form %>% 
+    mutate(
+      is_trans_column_invalid_or_missing = is_invalid_na_or_n(trans) |
+        (trans == 1 &
+           (is_invalid_na_or_n(transsp) | transsp == 0 |
+              is_invalid_or_na(bhelm) | is_invalid_or_na(mhelm) |
+              is_invalid_or_na(tbsbelt) | is_invalid_or_na(asbelt) | 
+              is_invalid_or_na(othtrans) |
+              (transsp == 6 & (is_n(bhelm) | bhelm == 0)) |
```

Comment:
I would put bhelm == 0 next to is_invalid_or_na(bhelm) instead of is_n(bhelm).

bhelm should never be 0. It is either N when transsp == 6. Otherwise it should be 1 or 2 (Yes or No). 0 always means that something wasn't chosen when it should have been.